api/src/main/java/com/epam/pipeline/manager/user/UserManager.java

```diff
@@ -83,10 +88,47 @@
     @Autowired
     private DataStorageValidator storageValidator;
 
-    @Transactional(propagation = Propagation.REQUIRED)
+    @Autowired
+    private FolderManager folderManager;
+
+    @Value("${storage.user.home.auto:false}")
+    private boolean shouldCreateDefaultHome;
+
+    @Value("${storage.user.home.template}")
+    private  String defaultUserStorageTemplateName;
+
     public PipelineUser createUser(String name, List<Long> roles,
                                    List<String> groups, Map<String, String> attributes,
                                    Long defaultStorageId) {
+        final PipelineUser newUser = createUser(name, roles, groups, attributes);
+        if (defaultStorageId != null) {
+            newUser.setDefaultStorageId(defaultStorageId);
+            userDao.updateUser(newUser);
+        } else if (shouldCreateDefaultHome) {
+            final AbstractDataStorage defaultStorage = createUserDefaultFolder(newUser).getStorages().get(0);
+            newUser.setDefaultStorageId(defaultStorage.getId());
+            userDao.updateUser(newUser);
+        }
+        return newUser;
+    }
+
+    private Folder createUserDefaultFolder(final PipelineUser user) {
+        final Folder folder = new Folder();
```

Comment:
Instead of getting `parentId` from template, let's get it from `SystemPreferences` and set it to `folder` here

Reply:
Done